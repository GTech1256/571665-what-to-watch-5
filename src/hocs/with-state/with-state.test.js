import React from "react";
import renderer from "react-test-renderer";
import PropTypes from "prop-types";
import {filmType} from "../../types";
import filmMock from "../../mocks/film";
import withState from "./with-state";

const MockComponent = ({state: {film}, onFilmClick}) => <button onClick={onFilmClick}>{film.name}</button>;

MockComponent.propTypes = {
  state: PropTypes.exact({
    film: PropTypes.exact(filmType).isRequired
  }).isRequired,
  onFilmClick: PropTypes.func.isRequired
};

const MockComponentWrapped = withState(
    {film: filmMock},
    () => ({
      onFilmClick() {}
    })
)(MockComponent);

it(`withState is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped />
  )).toJSON();

  expect(tree).toMatchSnapshot();
});
