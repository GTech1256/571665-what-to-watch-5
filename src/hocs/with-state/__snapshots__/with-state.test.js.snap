// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withState is rendered correctly 1`] = `
<button
  onClick={[Function]}
>
  The Grand Budapest Hotel
</button>
`;
